templates/prometheus.go

```diff
@@ -45,5 +53,26 @@ var PrometheusTemplate = promv1.Prometheus{
 			}},
 		},
 		Resources: utils.GetResourceRequirements("prometheus"),
+		Web:       PrometheusWebSpecTemplate.DeepCopy(),
+	},
+}
+
+var locSecretRef = corev1.LocalObjectReference{
+	Name: PrometheusServiceSecretName,
+}
+
+var PrometheusWebSpecTemplate = &promv1.WebSpec{
+	TLSConfig: &promv1.WebTLSConfig{
+		KeySecret: corev1.SecretKeySelector{
+			LocalObjectReference: locSecretRef,
```

Comment:
```suggestion
			LocalObjectReference: corev1.LocalObjectReference{Name: PrometheusServiceSecretName}
```
Or even better 
```suggestion
			LocalObjectReference: corev1.LocalObjectReference{Name: ""}
```
Then in the reconcile function, you can do
```
r.prometheus.Spec.Web.TLSConfig.KeySecret.LocalObjectReference.Name = PrometheusServiceSecretName
```

Which will allow you to keep `PrometheusServiceSecretName` on the controller where it makes more sense.

But either option is ok.

Reply:
Chose the second option (`Or even better` part), but didn't add any `LocalObjectReference` part since it is not a reference/pointer Go should create an empty defaults into it.
Added an assertion in the test to make sure all the values we set are properly propagated to prometheus instance.

Reply:
Good observation about not being a pointer, make the template even simpler 💯